src/index.coffee

```diff
@@ -6,45 +6,39 @@ BrowserTransport = require './transports/browser-transport'
 CliTransport = require './transports/cli-transport'
 Logger = require './logger'
 
-
-getGlobalScope = ->
-  if typeof global != 'undefined'
-    return global
-
-  return window
-
-
-isBrowser = (globalScope) ->
-  return (
-    globalScope.window?.document?.createElement and
-    globalScope.window?.name != 'nodejs'
-  )
+{ getGlobalScope, isBrowser } = require './utils'
+{ setupTransport } = require './setup-transport'
 
 
 module.exports =
   LogLevels: LogLevels
   
   Logger: Logger
 
+  BrowserTransport: BrowserTransport
+  CliTransport: CliTransport
+
   create: (options = {}) ->
     globalScope = getGlobalScope()
     browser = isBrowser(globalScope)
 
     if browser
-      transport = new BrowserTransport(globalScope.console)
-      transport.setLogLevelPrefixes(prefixes.browser)
-      transport.setLogLevelStyles(colors.browser)
+      output = globalScope.console
+      transportSource = BrowserTransport
+
     else
-      transport = new CliTransport(globalScope.process.stderr)
-      transport.setLogLevelPrefixes(prefixes.cli)
-      transport.setLogLevelColors(colors.cli)
+      output = globalScope.process.stderr
+      transportSource = CliTransport 
 
-    logger = new Logger(transport, options)
+    transport = if options.transport then new options.transport(output) else new transportSource(output)
+    transportWithSetup = setupTransport(transport)
+    logger = new Logger(transportWithSetup, options)
     nextConsole = logger.extendConsole(globalScope.console)
-    return nextConsole
+    return [ nextConsole, transport ]
```

Comment:
It does not seem like a good idea to expose the transport instance this way. If the user wants to have control over the instance, they can pass in their own via options.

Reply:
I think it's useful to expose it since you want to have the ability to switch to your own transport in `Logger#switchTransport`. This own transport might want to call the default transport that was passed to the logger so you don't get circular calls.